### ISSUE TITLE
Reorganise code to have a tool-agnostic core

### ISSUE THREAD

**Issue by pmonks** [noisesmith/lein-git-info-edn#7]
(from Slack conversation)

@noisesmith: I think, aesthetically, there should be a "wrapper" ns so that the deps edn code doesn't have to call code that looks like it belongs to leiningen or better yet, the leiningen and deps.edn entrypoints could each be a slim front over the actual impl.

**Comment by pmonks** [comment]
My proposal would be to move the non- lein/deps.edn -specific code into a `git-info-edn.core` namespace, then have one namespace each for the leiningen and deps.edn front-ends.  These would be `leiningen.git-info-edn` (required for backwards compatibility and for the plugin to be recognised by Leiningen) and `deps-edn.git-info-edn`, respectively.  It's slightly unwieldy to have to have all three namespaces in completely separate files **and** folders, but leiningen requires a specific location for plugin code (plus it's good to preserve backwards compatibility), and there's symmetry in placing the deps.edn entry point as described above (even though it could go just about anywhere).

It would also be ideal to rename the GitHub project to `noisesmith/git-info-edn`, though that can't be achieved with a PR...

Thoughts, @noisesmith?

**Comment by noisesmith** [comment]
usually I'm not a fan of .core namespaces, ideally it could have a more meaningful name

what about

```
src/
     noisesmith/
                      git_info.clj
     leiningen/
                      git_info_edn.clj
     deps_edn/
                      git_info_edn.clj
```

which  leads to

noisesmith.git-info/git-info-edn
leiningen.git-info-edn/git-info-edn
deps-edn.git-info-edn/-main

as the various entry points that would be defined

**Comment by noisesmith** [comment]
perhaps there's no need for a deps_edn root, and deps.edn could just call the -main defined in noisesmith.git-info/git-info-edn

**Comment by pmonks** [comment]
Yeah merging the deps.edn `-main` fn into the `noisesmith/git-info` ns is absolutely feasible.  It just slightly blurs the line between the core logic and the deps.edn specific bits (which admittedly is little more than a one liner!).  It also doesn't have quite the symmetry of your first proposal, for whatever that's worth.
